internal/handlers/common/update.go

```diff
@@ -236,6 +244,67 @@ func processPopFieldExpression(doc *types.Document, update *types.Document) (boo
 	return changed, nil
 }
 
+// processRenameFieldExpression changes document according to $rename operator.
+// If the document was changed it returns true.
+func processRenameFieldExpression(doc *types.Document, update *types.Document) (bool, error) {
+	renameExpression := update.SortFieldsByKey()
+
+	var changed bool
+
+	for _, key := range renameExpression.Keys() {
+		renameRawValue, err := renameExpression.Get(key)
+		if err != nil {
+			// if $rename field does not exist, don't change anything
+			continue
+		}
+
+		if key == "" || renameRawValue == "" {
+			return changed, NewWriteErrorMsg(ErrEmptyName, "An empty update path is not valid.")
+		}
+
+		// this is covered in extractValueFromUpdateOperator
```

Comment:
Hmm, I don't understand this comment. What does it mean?
Also, I see codecov complains that the next few lines are not covered.

Reply:
We don't need to check if that statement is `ok` and if `renameRawValue` underlying type is the string, as we covered it in `extractValueFromUpdateOperator` (the same code is now in `validateRenameExpression`). Any error related to value type validation will be returned in that function and we don't need to handle this another time. I can rephrase it to something else like "The correctness of this type assertion is made in `validateRenameExpresssion`" or smth like that